src/Amalgam/evaluablenode/EvaluableNodeManagement.cpp

```diff
@@ -736,6 +730,75 @@ void EvaluableNodeManager::NonCycleModifyLabelsForNodeTree(EvaluableNode *tree,
 	}
 }
 
+void EvaluableNodeManager::MarkAllReferencedNodesInUse(bool set_in_use)
+{
+#ifdef MULTITHREAD_SUPPORT
+	size_t reference_count = nodesCurrentlyReferenced.size();
+	//heuristic to ensure there's enough to do to warrant the overhead of using multiple threads
+	if(reference_count > 1 && firstUnusedNodeIndex / reference_count >= 2000)
```

Comment:
How do you feel about lack of parens here? I'm with "operator precedence is obvious" but I would lean more towards parens.